jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/thread/ReservedThreadExecutorTest.java

```diff
@@ -216,38 +213,18 @@ public void testReservedIdleTimeoutWithOneReservedThread() throws Exception
     {
         long idleTimeout = 500;
         _reservedExecutor.stop();
-        _reservedExecutor.setIdleTimeout(idleTimeout, TimeUnit.MILLISECONDS);
+        _reservedExecutor.setIdleTimeout(idleTimeout, MILLISECONDS);
         _reservedExecutor.start();
 
         assertThat(_reservedExecutor.tryExecute(NOOP), is(false));
         Thread thread = _executor.startThread();
         assertNotNull(thread);
-        waitForAvailable(1);
-
-        Thread.sleep(2 * idleTimeout);
-
-        waitForAvailable(0);
+        waitAtMost(5, SECONDS).until(_reservedExecutor::getAvailable, is(1));
```

Comment:
This is going to be flaky; awaitility may not see the transition to 1.

Reply:
CI should not pause for an entire idle timeout period.... if it does, I think more tests would fail.   So I think you are right technically, but I'm hoping that in practise it is fine.... or at least better than now.   Thoughts?

Reply:
Okay, there's a natural "long" delay between the two waits, so that should work fine.

Maybe it's worth a comment?

Reply:
added